dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -78,7 +87,21 @@ sealed class TypedColumn[T, U](
     *
     * apache/spark
     */
-  def =!=(other: U): TypedColumn[T, Boolean] = (self.untyped =!= lit(other).untyped).typed
+  def =!=(other: U): TypedColumn[T, Boolean] = withExpr {
+    Not(equalsTo(lit(other)).expr)
+  }.typed
+
+  /** True if the current expression is an Option and it's None.
+    *
+    * apache/spark
+    */
+  def isNone(implicit isOption: U <:< Option[_]): TypedColumn[T, Boolean] = self.untyped.isNull.typed
+
+  /** True if the current expression is an Option and it's not None.
+    *
+    * apache/spark
+    */
+  def isNotNone(implicit isOption: U <:< Option[_]): TypedColumn[T, Boolean] = self.untyped.isNotNull.typed
```

Comment:
@OlivierBlanvillain Let me get to this! I will add a test here. And rebase with master. Thank you for taking a look. Life has been very hectic lately, I hope things will stabilize soon.